pkg/queue/rabbitmq/internal/event/router.go

```diff
@@ -171,16 +181,44 @@ func (cer *CommonEventRouter) getSubscriber(pin string, subscriberType internal.
 	return result, nil
 }
 
+func (cer *CommonEventRouter) findSubscriber(pin string) internal.Subscriber {
+	cer.mutex.RLock()
+	defer cer.mutex.RUnlock()
+
+	if existing, ok := cer.subscribers[pin]; ok {
+		return existing
+	}
+	return nil
+}
+
 func (cer *CommonEventRouter) getSender(pin string) *CommonEventSender {
 	queueConfig := cer.config.Queues[pin] // get queue by pin
 	var result *CommonEventSender
-	if _, ok := cer.senders[pin]; ok {
-		result = cer.senders[pin]
+	result = cer.findSender(pin)
+	if result != nil {
 		return result
 	}
+
+	cer.mutex.Lock()
+	defer cer.mutex.Unlock()
```

Comment:
Could we lock mutex before first check to execute check and create new sender atomically?

Reply:
We call RLock in findSender\findSubscriber methods. Then if the sender\subscriber is not initialized we take Lock

Reply:
I understand your idea:
* when sender / subscriber is already initialized than read lock is enough
* when sender / subscriber was initializing first time than we check existence under read lock. If entity isn't existed then double check and create new instance under write lock